wire-ios-sync-engine/Source/Synchronization/RecurringActionService.swift

```diff
@@ -0,0 +1,96 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+struct RecurringAction {
+
+    typealias Action = () -> Void
+
+    let id: String
+    let interval: TimeInterval
+    let perform: Action
+
+}
+
+protocol RecurringActionServiceInterface {
+
+    func performActionsIfNeeded()
+    func registerAction(_ action: RecurringAction)
+
+}
+
+class RecurringActionService: RecurringActionServiceInterface {
+
+    var userDefaultsName = "com.wire.recurringAction."
```

Comment:
I think it would be better to inject in the user defaults instance for test purposes rather than modifying the name used to create the user defaults.

Reply:
How do you suggest injecting the user defaults instance, since we should have those instances per action, and not per `RecurringActionService`? Each action has its own time interval. `RecurringActionService` will contain several actions. 
If you think we should inject user defaults instance per action, it looks like we should extend `RecurringAction`:
```
struct RecurringAction {

    typealias Action = () -> Void

    let id: String
    let interval: TimeInterval
    let perform: Action

    /// New
   let storage: UserDefaults
}
```

Reply:
I applied your suggestion. Please check it out.